boilerplate/app/models/AuthenticationStore.ts

```diff
@@ -44,6 +44,11 @@ export const AuthenticationStoreModel = types
       store.authPassword = ""
     },
   }))
+  .preProcessSnapshot((snapshot) => {
+    // remove sensitive data from snapshot
+    const { authToken, authPassword, ...rest } = snapshot // eslint-disable-line @typescript-eslint/no-unused-vars
+    return rest
+  })
```

Comment:
I'm not as familiar with mobx, but won't this completely remove `authToken` and `authPassword` from being able to be used on the model?

Reply:
I'm not that familiar with mobx yet either but from [the docs](https://mobx-state-tree.js.org/overview/hooks):

> Before creating an instance or applying a snapshot to an existing instance, this hook is called to give the option to transform the snapshot before it is applied. The hook should be a pure function that returns a new snapshot. This can be useful to do some data conversion, enrichment, property renames, etc. This hook is not called for individual property updates. 

So I believe it just removes it from the snapshot that's being applied/generated and it will still work on the instance. I'm actually wondering if it shouldn't be `postProcessSnapshot` now. Maybe @jamonholmgren can chime in?

Reply:
I think @yulolimum mentioned this method, see https://infinitered.slack.com/archives/C0X0P5LP7/p1666302551358189?thread_ts=1666302363.164549&cid=C0X0P5LP7

Reply:
That works for me then!

Reply:
I think it would be worth adding a more specific comment to cue the developer to think about what they should do on their project

```suggestion
  .preProcessSnapshot((snapshot) => {
    // remove sensitive data from snapshot to avoid secrets
    // being stored in AsyncStorage in plain text if backing up store
    const { authToken, authPassword, ...rest } = snapshot // eslint-disable-line @typescript-eslint/no-unused-vars
    
    // see the following for strategies to consider storing secrets on device
    // https://reactnative.dev/docs/security#storing-sensitive-info
    
    return rest
  })
```